### PR TITLE
Allow checking whether all expectations have been met

### PR DESCRIPTION
**Pull request by FlorianLoch**
As suggested and outlined in #10

### CODE REVIEW

**Comment by FlorianLoch** [comment]
In case you are wondering why the commits are so old: I made this PR initially against `golang/mock`.

File: gomock/callset.go [hunk 1]
```diff
@@ -111,3 +131,19 @@ func (cs callSet) Failures() []*Call {
 	}
 	return failures
 }
+
+// AllExpectedCallsSatisfied returns true in case all expected calls in this callSet are satisfied.
+func (cs callSet) AllExpectedCallsSatisfied() bool {
```
Comment (sywhang) [inline gomock/callset.go:136]: re: naming, this is a mouthful, can we call this `Satisfied` instead?
```suggestion
func (cs callSet) Satisfied() bool {
```

File: gomock/controller.go [hunk 2]
```diff
@@ -296,3 +296,9 @@ func unwrapTestReporter(t TestReporter) TestReporter {
 	}
 	return tr
 }
+
+// AllExpectedCallsSatisfied returns true in case all expected calls bound to this Controller are satisfied.
+// Calling Finish is then guaranteed to not fail due to missing calls.
+func (ctrl *Controller) AllExpectedCallsSatisfied() bool {
```
Comment (sywhang) [inline gomock/controller.go:302]: Same thing here as above 
```suggestion
func (ctrl *Controller) Satisfied() bool {
```

File: gomock/controller.go [hunk 3]
```diff
@@ -296,3 +296,9 @@ func unwrapTestReporter(t TestReporter) TestReporter {
 	}
 	return tr
 }
+
+// AllExpectedCallsSatisfied returns true in case all expected calls bound to this Controller are satisfied.
```
Comment (sywhang) [inline gomock/controller.go:300]: let's move this to be next to other exported functions, closer to top of the file.

**Review by sywhang — COMMENTED** [review]
mostly looks good to me, left a couple of suggestions re: naming and method ordering.

File: gomock/controller.go (hunk 3 above)
Comment (FlorianLoch) [reply gomock/controller.go:300]: 👍

File: gomock/controller.go (hunk 2 above)
Comment (FlorianLoch) [reply gomock/controller.go:302]: 👍

File: gomock/callset.go (hunk 1 above)
Comment (FlorianLoch) [reply gomock/callset.go]: 👍

File: gomock/controller.go [hunk 4]
```diff
@@ -231,6 +231,12 @@ func (ctrl *Controller) Finish() {
 	ctrl.finish(false, err)
 }
 
+// Satisfied returns true in case all expected calls bound to this Controller are satisfied.
```
Comment (sywhang) [inline gomock/controller.go:234]:
```suggestion
// Satisfied returns whether all expected calls bound to this Controller have been satisfied.
```

**Review by sywhang — APPROVED** [review]

**Pull request merged**